pkg/stub/handler.go

```diff
@@ -2,50 +2,97 @@ package stub
 
 	"github.com/operator-framework/operator-sdk/pkg/sdk"
 
-	"github.com/sirupsen/logrus"
-
-	appsv1 "k8s.io/api/apps/v1"
-	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/errors"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/runtime/schema"
-	"k8s.io/apimachinery/pkg/util/intstr"
 )
 
 func NewHandler() sdk.Handler {
-	return &Handler{}
+	return &Handler{
+		manifestFactory: manifests.NewFactory(),
+	}
 }
 
 type Handler struct {
-	// Fill me
+	manifestFactory *manifests.Factory
 }
 
 func (h *Handler) Handle(ctx context.Context, event sdk.Event) error {
 	if event.Deleted {
 		return nil
 	}
 	switch o := event.Object.(type) {
-	case *v1alpha1.ClusterIngress:
-		err := sdk.Create(newRouterDaemonSet(o))
-		if err != nil && !errors.IsAlreadyExists(err) {
-			logrus.Errorf("Failed to create daemonset: %v", err)
-			return err
-		}
-		err = sdk.Create(newRouterService(o))
-		if err != nil && !errors.IsAlreadyExists(err) {
-			logrus.Errorf("Failed to create service: %v", err)
-			return err
-		}
+	case *ingressv1alpha1.ClusterIngress:
+		return h.syncIngressUpdate(o)
+	}
+	return nil
+}
+
+func (h *Handler) syncIngressUpdate(ci *ingressv1alpha1.ClusterIngress) error {
+	ns, err := h.manifestFactory.RouterNamespace()
+	if err != nil {
+		return fmt.Errorf("couldn't build router namespace: %v", err)
+	}
+	err = sdk.Create(ns)
+	if err != nil && !errors.IsAlreadyExists(err) {
+		return fmt.Errorf("couldn't create router namespace: %v", err)
+	}
+
+	sa, err := h.manifestFactory.RouterServiceAccount()
+	if err != nil {
+		return fmt.Errorf("couldn't build router service account: %v", err)
+	}
+	err = sdk.Create(sa)
+	if err != nil && !errors.IsAlreadyExists(err) {
+		return fmt.Errorf("couldn't create router service account: %v", err)
+	}
+
+	cr, err := h.manifestFactory.RouterClusterRole()
+	if err != nil {
+		return fmt.Errorf("couldn't build router cluster role: %v", err)
 	}
+	err = sdk.Create(cr)
+	if err != nil && !errors.IsAlreadyExists(err) {
+		return fmt.Errorf("couldn't create router cluster role: %v", err)
+	}
+
+	crb, err := h.manifestFactory.RouterClusterRoleBinding()
+	if err != nil {
+		return fmt.Errorf("couldn't build router cluster role binding: %v", err)
+	}
+	err = sdk.Create(crb)
+	if err != nil && !errors.IsAlreadyExists(err) {
+		return fmt.Errorf("couldn't create router cluster role binding: %v", err)
+	}
+
+	ds, err := h.manifestFactory.RouterDaemonSet(ci)
+	if err != nil {
+		return fmt.Errorf("couldn't build daemonset: %v", err)
+	}
+	err = sdk.Create(ds)
+	if err != nil && !errors.IsAlreadyExists(err) {
+		return fmt.Errorf("failed to create daemonset: %v", err)
+	}
+
+	service, err := h.manifestFactory.RouterServiceCloud(ci)
+	if err != nil {
+		return fmt.Errorf("couldn't build service: %v", err)
+	}
+	err = sdk.Create(service)
+	if err != nil && !errors.IsAlreadyExists(err) {
+		return fmt.Errorf("failed to create service: %v", err)
+	}
+
```

Comment:
Maybe later this could be compacted in a sdkCreate function

Reply:
Yeah, lots of boilerplate here